pkg/cache/cache.go

```diff
@@ -136,6 +135,55 @@ func (sc *schedulerCache) WaitForCacheSync(stopCh <-chan struct{}) bool {
 		sc.consumerInformer.Informer().HasSynced)
 }
 
+// nonTerminatedPod selects pods that are non-terminal (scheduled and running).
+func nonTerminatedPod(pod *v1.Pod) bool {
+	if pod.Status.Phase == v1.PodSucceeded || pod.Status.Phase == v1.PodFailed {
+		return false
+	}
+	return true
+}
+
+func (sc *schedulerCache) addPodToNode(pi *PodInfo) {
+	if pi.Phase != v1.PodRunning {
+		// only add running pod to node
+		return
+	}
+	if pi.Nodename == "" {
+		// the pod is not assigned to any node, do nothing
+		return
+	}
+
+	_, exist := sc.nodes[pi.Nodename]
+	if !exist {
+		// no related node info for the pod, do nothing
+		// this pod will be added into related node when the node is inserted into cache
+		return
+	}
+
+	sc.nodes[pi.Nodename].Pods = append(sc.nodes[pi.Nodename].Pods, pi)
+	sc.nodes[pi.Nodename].Used.Add(pi.Request)
+	sc.nodes[pi.Nodename].Idle = sc.nodes[pi.Nodename].Allocatable.Clone()
+	sc.nodes[pi.Nodename].Idle.Sub(sc.nodes[pi.Nodename].Used)
+}
+
+func (sc *schedulerCache) removePodFromNode(pi *PodInfo) {
+	_, exist := sc.nodes[pi.Nodename]
+	if !exist {
+		// no related node info for the pod, do nothing
+		return
+	}
+
+	for index, p := range sc.nodes[pi.Nodename].Pods {
+		if pi.Name == p.Name {
+			sc.nodes[pi.Nodename].Pods = append(sc.nodes[pi.Nodename].Pods[:index], sc.nodes[pi.Nodename].Pods[index+1:]...)
+			sc.nodes[pi.Nodename].Used.Sub(pi.Request)
+			sc.nodes[pi.Nodename].Idle = sc.nodes[pi.Nodename].Allocatable.Clone()
```

Comment:
should not clone `Allocatable` every time, the allocatable of node will not change.